### DISCUSSION TITLE
How to stop propagation of the HTML overlay click event?

### DISCUSSION THREAD

**Discussion by NameFILIP** [visgl/deck.gl#6252] [Q&A] [answered]
👋 I'm having troubles implementing a somewhat simple behavior:

* I have an IconLayer that is rendering a list of items. 
* Each item can be selected if clicked. 
* It can be de-selected when clicking somewhere else on the map (not another item). 
* When an item is selected an HTML Overlay is shown that displays some information about the selected item.

How can I click on the HTML Overlay so that the click event is not delivered to the map (I want the item to still be selected after this click)?

See the example: https://codesandbox.io/s/react-map-gl-osm-and-digital-globe-example-forked-nlpui?file=/src/index.js
Notice that currently, after selecting an item and then clicking on the shown HTML Overlay the item is de-selected.

I tried calling `event.stopPropagation()` in the `HTMLOverlayItem`'s `onClick`, but turns out the `DeckGL`'s `event.srcEvent.type` is `pointerup`, so it is not aware that `click` propagation was stopped.

What is the best way to solve this?

**Comment by NameFILIP** [comment, marked as answer]
I found a workaround. It's very non-obvious, but works well:
```
onPointerUp={(event) => event.preventDefault()}
```
on the HTML elements and
```
if (event.srcEvent.defaultPrevented) {
  return true;
}
```
in the Deck.GL's `onClick` event handlers.

Similarly, for `onHover` instead of `onPointerUp` use `onMouseMove`:
```
onMouseMove={(event) => event.preventDefault()}
```
See the full example: https://codesandbox.io/s/react-map-gl-osm-and-digital-globe-example-forked-pm9sf

**Comment by efumagal** [comment]
I wonder if there's a way to prevent all events not to be delivered on the map.
In the examples like https://deck.gl/examples/tile-layer, the events in the overlay are not propagated.
How was that done there ?

**Comment by efumagal** [comment]
In my case it was easier than I thought, thanks to the people on the Slack channel I found out that I just needed to put the overlay outside of the deck.gl component.

**Reply by kgreenek** [reply]
This! Thank you!

**Reply by StoneTapeStudios** [reply]
Dude thank you, took me way too long to find this answer 🙏😎
